backend/coins/btc/handlers/handlers.go

```diff
@@ -580,3 +583,102 @@ func (handlers *Handlers) postSetTxNote(r *http.Request) (interface{}, error) {
+	if !ok {
+		return signingResponse{Success: false, ErrorMessage: "Must be an ETH based account"}, nil
+	}
+	signature, err := ethAccount.SignMsg(signInput)
+	if errp.Cause(err) == keystore.ErrSigningAborted {
+		return signingResponse{Success: false, Aborted: true}, nil
+	}
+	if err != nil {
+		handlers.log.WithError(err).Error("Failed to sign message")
+		result := signingResponse{Success: false, ErrorMessage: err.Error()}
+		return result, nil
+	}
+	return signingResponse{
+		Success:   true,
+		Signature: signature,
+	}, nil
+}
+
+func (handlers *Handlers) postEthSignTypedMsg(r *http.Request) (interface{}, error) {
+	var args struct {
+		ChainId uint64 `json:"chainId"`
+		Data    string `json:"data"`
+	}
+	if err := json.NewDecoder(r.Body).Decode(&args); err != nil {
+		return signingResponse{Success: false, ErrorMessage: errp.WithStack(err).Error()}, nil
+	}
+	ethAccount, ok := handlers.account.(*eth.Account)
+	if !ok {
+		return signingResponse{Success: false, ErrorMessage: "Must be an ETH based account"}, nil
+	}
+	signature, err := ethAccount.SignTypedMsg(args.ChainId, args.Data)
+	if errp.Cause(err) == keystore.ErrSigningAborted {
+		return signingResponse{Success: false, Aborted: true}, nil
+	}
+	if err != nil {
+		handlers.log.WithError(err).Error("Failed to sign typed data")
+		result := signingResponse{Success: false, ErrorMessage: err.Error()}
+		return result, nil
+	}
+	return signingResponse{
+		Success:   true,
+		Signature: signature,
+	}, nil
+}
+
+// For handling dapp transaction requests through Wallet Connect which can either request tx sign or tx send
+// The `json:"send"` bool specifies whether a tx should be only signed (return signature) or signed and broadcast (return tx hash)
+// ChainId is needed to allow signing all supported EVM networks via the BBApp
+func (handlers *Handlers) postEthSignWalletConnectTx(r *http.Request) (interface{}, error) {
+	var args struct {
+		Send    bool                  `json:"send"`
+		ChainId uint64                `json:"chainId"`
+		Tx      eth.WalletConnectArgs `json:"tx"`
+	}
+	type response struct {
+		Success bool   `json:"success"`
+		RawTx   string `json:"rawTx"`
+		TxHash  string `json:"txHash"`
+	}
+	if err := json.NewDecoder(r.Body).Decode(&args); err != nil {
+		return signingResponse{Success: false, ErrorMessage: errp.WithStack(err).Error()}, nil
+	}
+	ethAccount, ok := handlers.account.(*eth.Account)
+	if !ok {
+		return signingResponse{Success: false, ErrorMessage: "Must be an ETH based account"}, nil
+	}
+	txHash, rawTx, err := ethAccount.EthSignWalletConnectTx(args.Send, args.ChainId, args.Tx)
+	if errp.Cause(err) == keystore.ErrSigningAborted {
+		return signingResponse{Success: false, Aborted: true}, nil
+	}
+	if err != nil {
+		handlers.log.WithError(err).Error("Failed to send transaction")
+		result := signingResponse{Success: false, ErrorMessage: err.Error()}
+		if err.Error() == errors.ErrInsufficientFunds.Error() {
+			result.ErrorCode = err.Error()
```

Comment:
is this error code even used in the frontend or anywhere?

Reply:
Good catch, it isn't right now and the ErrorMessage should be sufficient in any case since it contains the same err.Error(), removed.